docs/developing_iree/profiling_with_tracy.md

```diff
@@ -162,42 +164,81 @@ If connecting doesn't work:
-generated code tends to run on a thread named `cpu0`, which is actually a thread
-name and unrelated to `CPU0` from the systrace view.
-
-Once you click a zone, you should see a line-by-line or
-instruction-by-instruction view annotated with the percentage of time spent. If
-you don't see the percentages, then again that would point to sampling not
-working, refer to the above steps.
+You should then start seeing a profile. The initial view should look like this:
+
+![Tracy initial view, normal
+case](https://gist.githubusercontent.com/bjacob/ff7dec20c1dfc7d0fc556cc7275bca9a/raw/fe4e22ca0301ebbfd537c47332a4a2c300a417b3/tracy_initial_view.jpeg)
+
+Before going further, take a second to check that your recorded profile data has
+all the data that it should have. Permissions issues, as discussed above, could
+cause it to lack "sampling" or "CPU data" information. For example, here is what
+he initial view looks like when one forgot to run the profiled program as root
+on Desktop Linux (where running as root is required, as explained above):
+
+![Tracy initial view, permissions
+issue](https://gist.githubusercontent.com/bjacob/ff7dec20c1dfc7d0fc556cc7275bca9a/raw/fe4e22ca0301ebbfd537c47332a4a2c300a417b3/tracy_permissions_issue.jpeg)
+
+Notice how the latter screenshot is lacking the following elements:
+* No 'CPU data' header on the left side, with the list of all CPU cores. The
+  'CPU usage' graph is something else.
+* No 'ghost' icon next to the 'Main thread' header.
+
+When running into any of the above issues, refer to the above Permissions
+section. Look for any interesting `stderr` message (in the profiled program's
+terminal).
+
+Click the 'Statistics' button at the top. It will open a window like this:
+
+![Tracy statistics
+window](https://gist.githubusercontent.com/bjacob/ff7dec20c1dfc7d0fc556cc7275bca9a/raw/fe4e22ca0301ebbfd537c47332a4a2c300a417b3/tracy_statistics.jpeg)
+
+See how the above screenshot has two radio buttons at the top: 'Instrumentation'
+and 'Sampling'. At this point, if you don't a the 'Sampling' radio button, you
+need to resolve that first, as discussed above about possible permissions
+issues.
+
+These 'Instrumentation' and 'Sampling' statistics correspond the two kinds of
+data that Tracy collects about your program. In the Tracy main view, they
+correspond, respectively, to 'normal' and 'ghost' zones. For each thread, the
+ghost icon toggles the view between these two kinds of zones. 'Normal' zones are
+explicitly recorded by the instrumented program, with timestamps. The stacks
+shown are stacks of Tracy instrumentation zones, not real call stacks. 'Ghost'
+zones are obtained by Tracy sampling the thread's real call stack at a frequency
+of typically roughly 10 kHz. Only ghost zones have the data to show the time
+spent on a line-by-line or instruction-by-instruction basis. That's why when we
+actually click on a zone, we get a very different view depending on whether it's
+a normal or a ghost zone.
```

Comment:
If this gets more comprehensive, I'd suggest referring people to the Tracy manual for a general introduction to Tracy and then focusing our docs on what is noteworthy about IREE (e.g. which threads and event patterns to look for).

Reply:
good point, linked and trimmed this text.